src/Practitioner/Settings/index.js

```diff
@@ -1,215 +1,289 @@
-    }
-
-    return (<li className={classes.navItemContainer}>
-        <ButtonBase href={props.href} target="blank" onClick={() => { handleClick(props.id) }} className={classes.navItem}>
-            {props.icon}
-            <span>{props.text}</span>
-        </ButtonBase>
-    </li>)
-})
-
-const BodyRouter = observer((props) => {
-    const { practitionerUIStore } = useStores();
-    const { t } = useTranslation('translation');
-    const classes = useStyles();
-    return (
-        <div className={classes.body}>
-            <Typography className={classes.header} variant="h2">{t(`coordinator.settingsPage.${practitionerUIStore.settingsTab}`)}</Typography>
-            <div className={classes.bodyContent}>
-                {practitionerUIStore.settingsTab === "language" && <Box padding="2em 0"><Language
-                    selectedBackgroundColor={Colors.textDarkGray}
-                    selectedTextColor="white"
-                    defaultBackgroundColor={Colors.lightgray}
-                    defaultTextColor={Colors.textDarkGray}
-                    defaultBorderColor={Colors.gray}
-                    selectedBorderColor={Colors.gray}
-
-                />
-                </Box>}
-                {practitionerUIStore.settingsTab === "documents" && <Documents />}
-                {practitionerUIStore.settingsTab === "updatePassword" && <div className={classes.password}><PasswordReset /></div>}
-                {practitionerUIStore.settingsTab === "account" && <Profile />}
-                {practitionerUIStore.settingsTab === "patientInformation" && <div className={classes.patientInformation}>
-                    <Typography className={classes.topInfo} variant="body1">
-                        {t('coordinator.settingsPage.patientViewExplanation')}
-                    </Typography>
-                    <PatientInformation />
-                </div>}
-            </div>
-        </div>
+  const location = useLocation();
+  const { isMobile } = useWindowSize();
+
+  const classes = useStyles({
+    selected:
+      (props.to === 'documents' &&
+        location.pathname === '/settings' &&
+        !isMobile) ||
+      (props.to === location.pathname.split('/')[2] && props.to),
+    isLogout: props.isLogout,
+  });
 
-    )
-})
+  const logout = useLogout();
+
+  const LinkButton = (props) => (
+    <ButtonBase
+      component={props.to ? Link : 'button'}
+      to={props.to}
+      href={props.href}
+      target={props.href ? 'blank' : null}
+      onClick={props.onClick}
+      className={`${classes.navItem}`}
+    >
+      {props.icon}
+      <span>{props.text}</span>
+    </ButtonBase>
+  );
+
+  return (
+    <li className={classes.navItemContainer}>
+      <LinkButton
+        to={props.to && `/settings/${props.to}`}
+        icon={props.icon}
+        href={props.href}
+        text={props.text}
+        onClick={props.isLogout && logout}
+      />
+    </li>
+  );
+});
+
+const Routes = observer((props) => {
```

Comment:
Factor this out into its own file in this folder

Reply:
Done. Can be found in WrappedRoute.js in same Settings folder